kgforge/specializations/models/rdf_model.py

```diff
@@ -87,27 +89,30 @@ def _generate_context(self) -> Context:
         if document:
             return Context(document)
 
+    def _sparql(self, query) -> SPARQLResult:
+        return self.service.sparql(query)
+
     # Templates.
 
     def _template(self, type: str, only_required: bool) -> Dict:
-        try:
-            uri = self.service.types_to_shapes[type]
-        except KeyError:
-            raise ValueError("type '"+type+"' not found in "+self.source)
+        uri = self.get_shape_from_type(type)
         node_properties = self.service.materialize(uri)
         dictionary = parse_attributes(node_properties, only_required, None)
         return dictionary
 
-    # Validation.
+    def get_shape_from_type(self, type: str) -> URIRef:
+        if type not in self.service.types_to_shapes:
+            raise ValueError(f"Type {type} not found")
+        return self.service.types_to_shapes[type]
 
-    def schema_id(self, type: str) -> str:
-        try:
-            shape_iri = self.service.types_to_shapes[type]
-            return str(self.service.schema_source_id(shape_iri))
-        except KeyError:
-            raise ValueError("type not found")
+    def schema_id(self, type: str) -> URIRef:
```

Comment:
Why URIRef ?

Reply:
https://github.com/BlueBrain/nexus-forge/blob/master/kgforge/specializations/models/rdf/store_service.py#L103 it's a consequence of this line. The dictionary value was set to be a URIRef for class_to_shapes and this is used by _build_types_to_shapes to create types_to_shapes and get_shape_from_type gets it from it. So I just added types